lambdaclass/mirra_backend
# Send StatAffected messages when the energy of a unit changes

## Motivation

Closes https://github.com/lambdaclass/champions_of_mirra/issues/396
We need to send a message to the front when a unit's energy changes, in order to be able to implement each unit's energy bar.

## Summary of changes

Same as we do with the StatAffected message we use for a health modification, we now send a message when the energy changes.

## How to test it?

Start the application and run a battle from the client, or as explained in battle.md. Look in the log for a message such as this one: "[info] Dealing 41 damage to Uren-890 (875 -> 834). **Target energy recharge: 50**". The part marked in bold is the new one. 

If you want to check that the client is receiving the new message, you can:
1. Move the SocketTester to /lib/gateway, simulate a battle there and check that the :energy_recharge message is being logged.
2. Replicate the gateway.proto changes in the frontend project, go to the PlayOutSteps method in BattleManager.cs and add this case to the switch: 
`case Protobuf.Messages.Action.ActionTypeOneofCase.EnergyRegen:
      Debug.Log("Energy changed");
      break;
`
Run a battle, and see how "Energy changed" is printed in the console.

## Checklist
